tests/opentelemetry-test-utils/src/opentelemetry/test/test_base.py

```diff
@@ -130,6 +137,100 @@ def disable_logging(highest_level=logging.CRITICAL):
+            self.memory_metrics_reader.get_metrics_data().resource_metrics
+        )
+
+        all_metrics = []
+        for metrics in resource_metrics:
+            for scope_metrics in metrics.scope_metrics:
+                all_metrics.extend(scope_metrics.metrics)
+
+        return self.sorted_metrics(all_metrics)
+
+    @staticmethod
+    def sorted_metrics(metrics):
+        """
+        Sorts metrics by metric name.
+        """
+        return sorted(
+            metrics,
+            key=lambda m: m.name,
+        )
+
+    def assert_metric_expected(
+        self,
+        metric: Metric,
+        expected_data_points: Sequence[DataPointT],
+        est_value_delta: Optional[float] = 0,
+    ):
+        self.assertEqual(
+            len(expected_data_points), len(metric.data.data_points)
+        )
+        for expected_data_point in expected_data_points:
+            self.assert_data_point_expected(
+                expected_data_point, metric.data.data_points, est_value_delta
+            )
+
+    # pylint: disable=unidiomatic-typecheck
+    @staticmethod
+    def is_data_points_equal(
+        expected_data_point: DataPointT,
+        data_point: DataPointT,
+        est_value_delta: Optional[float] = 0,
+    ):
+        if type(expected_data_point) != type(data_point) or not isinstance(
+            expected_data_point, (HistogramDataPoint, NumberDataPoint)
+        ):
+            return False
+
+        values_diff = None
+        if isinstance(data_point, NumberDataPoint):
+            values_diff = abs(expected_data_point.value - data_point.value)
+        elif isinstance(data_point, HistogramDataPoint):
+            values_diff = abs(expected_data_point.sum - data_point.sum)
+
+            if (
+                expected_data_point.count != data_point.count
+                or expected_data_point.min != data_point.min
+                or expected_data_point.max != data_point.max
+            ):
+                return False
+
+        return (
+            values_diff <= est_value_delta
+            and expected_data_point.attributes == dict(data_point.attributes)
+        )
+
+    def assert_data_point_expected(
+        self,
+        expected_data_point: DataPointT,
+        data_points: Sequence[DataPointT],
+        est_value_delta: Optional[float] = 0,
+    ):
+        is_data_point_exist = False
+        for data_point in data_points:
+            if self.is_data_points_equal(
+                expected_data_point, data_point, est_value_delta
+            ):
+                is_data_point_exist = True
+                break
+
+        self.assertTrue(
+            is_data_point_exist,
+            msg=f"Data point {expected_data_point} does not exist",
+        )
+
+    @staticmethod
+    def create_number_data_point(value, attributes):
+        return NumberDataPoint(
+            value=value,
+            attributes=attributes,
+            start_time_unix_nano=0,
+            time_unix_nano=0,
+        )
+
```

Comment:
Should we also have `create_histogram_data_point`?

Reply:
Thanks
I will add support for histogram too